validator/node/node.go

```diff
@@ -503,18 +510,28 @@ func (c *ValidatorClient) registerRPCGatewayService(cliCtx *cli.Context) error {
 			"text/event-stream", &gwruntime.EventSourceJSONPb{},
 		),
 	)
-	muxHandler := func(h http.Handler, w http.ResponseWriter, req *http.Request) {
-		if strings.HasPrefix(req.URL.Path, "/api") {
-			http.StripPrefix("/api", h).ServeHTTP(w, req)
+	muxHandler := func(apiMware *apimiddleware.ApiProxyMiddleware, h http.HandlerFunc, w http.ResponseWriter, req *http.Request) {
+		// The validator gateway handler has some special logic as it serves two kinds of APIs
+		// and also the website for the Prysm Web UI.
+		if strings.HasPrefix(req.URL.Path, "/api/eth/") {
+			req.URL.Path = strings.Replace(req.URL.Path, "/api", "", 1)
+			// If the suffix has /eth/, we handle it with the standard API gateway middleware
+			// which will convert bytes values into hex strings as desired by the ETH standard API.
```

Comment:
Mentioning hex like this gives the illusion that this is the only thing the middleware will do for the keymanager API. This might be true now, but not in the future, which will render the comment outdated.